File: backend/app/api/api_v1/endpoints/cash_flow.py
"""
Endpoints REST para Fluxo de Caixa (Cash Flow) e Contas Bancárias

Fornece APIs para gerenciamento de:
- Contas Bancárias (CRUD)
- Movimentações Financeiras (CRUD)
- Transferências entre Contas
- Analytics e Projeções
- Relatórios Financeiros

Fase: 2.1 e 2.2 - Sprint Backend Cash Flow
Referência: roadmaps/ROADMAP_FINANCEIRO_INTEGRACAO.md
Autor: Sistema Orion ERP
Data: 2025-01-30
"""

from fastapi import APIRouter, Depends, HTTPException, Query, status
from sqlalchemy.orm import Session
from sqlalchemy import and_, or_, func, extract, case
from typing import List, Optional
from datetime import datetime, date, timedelta

from app.core.deps import get_current_user, get_db
from app.models.user import User
from app.models.cash_flow import BankAccount, CashFlowTransaction, TransactionType
from app.schemas.cash_flow import (
    # Bank Account
    BankAccountCreate,
    BankAccountUpdate,
    BankAccountResponse,
    AccountBalanceSummary,
    # Cash Flow Transaction
    CashFlowTransactionCreate,
    CashFlowTransactionUpdate,
    CashFlowTransactionResponse,
    TransferRequest,
    TransferResponse,
    # Analytics
    CashFlowSummary,
    CategorySummary,
    BalanceHistory,
    CashFlowProjection,
    CashFlowAnalytics,
    TransactionTypeEnum,
)

router = APIRouter()


# ============================================
# BANK ACCOUNTS ENDPOINTS
# ============================================

@router.post("/bank-accounts", response_model=BankAccountResponse, status_code=status.HTTP_201_CREATED)
def create_bank_account(
    account: BankAccountCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Criar nova conta bancária

    Cria uma nova conta bancária ou caixa para controle financeiro.
    """
    db_account = BankAccount(
        **account.dict(),
        workspace_id=current_user.workspace_id,
        current_balance=account.initial_balance or 0.0,
        created_by=current_user.id,
        updated_by=current_user.id
    )

    db.add(db_account)
    db.commit()
    db.refresh(db_account)

    return db_account


@router.get("/bank-accounts", response_model=List[BankAccountResponse])
def get_bank_accounts(
    skip: int = Query(0, ge=0),
    limit: int = Query(100, ge=1, le=1000),
    active_only: bool = Query(True, description="Retornar apenas contas ativas"),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Listar contas bancárias

    Retorna lista de contas bancárias do workspace com filtros opcionais.
    """
    query = db.query(BankAccount).filter(
        BankAccount.workspace_id == current_user.workspace_id
    )

    if active_only:
        query = query.filter(BankAccount.is_active == True)

    query = query.order_by(BankAccount.bank_name, BankAccount.created_at.desc())
    accounts = query.offset(skip).limit(limit).all()

    return accounts


@router.get("/bank-accounts/{account_id}", response_model=BankAccountResponse)
def get_bank_account(
    account_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Buscar conta bancária por ID

    Retorna detalhes de uma conta bancária específica.
    """
    account = db.query(BankAccount).filter(
        and_(
            BankAccount.id == account_id,
            BankAccount.workspace_id == current_user.workspace_id
        )
    ).first()

    if not account:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Bank account with id {account_id} not found"
        )

    return account


@router.patch("/bank-accounts/{account_id}", response_model=BankAccountResponse)
def update_bank_account(
    account_id: int,
    account_update: BankAccountUpdate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Atualizar conta bancária

    Atualiza informações de uma conta bancária. Saldo é atualizado apenas via transações.
    """
    db_account = db.query(BankAccount).filter(
        and_(
            BankAccount.id == account_id,
            BankAccount.workspace_id == current_user.workspace_id
        )
    ).first()

    if not db_account:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Bank account with id {account_id} not found"
        )

    update_data = account_update.dict(exclude_unset=True)
    for field, value in update_data.items():
        setattr(db_account, field, value)

    db_account.updated_by = current_user.id
    db_account.updated_at = datetime.utcnow()

    db.commit()
    db.refresh(db_account)

    return db_account


@router.delete("/bank-accounts/{account_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_bank_account(
    account_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Deletar conta bancária

    Remove uma conta bancária. Apenas contas sem transações podem ser deletadas.
    """
    db_account = db.query(BankAccount).filter(
        and_(
            BankAccount.id == account_id,
            BankAccount.workspace_id == current_user.workspace_id
        )
    ).first()

    if not db_account:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Bank account with id {account_id} not found"
        )

    # Verificar se há transações associadas
    transaction_count = db.query(func.count(CashFlowTransaction.id)).filter(
        CashFlowTransaction.account_id == account_id
    ).scalar()

    if transaction_count > 0:
        raise HTTPException(
            status_code=status.HTTP_400_BAD_REQUEST,
            detail=f"Cannot delete account with {transaction_count} transactions. Deactivate it instead."
        )

    db.delete(db_account)
    db.commit()

    return None


# ============================================
# CASH FLOW TRANSACTIONS ENDPOINTS
# ============================================

@router.post("/transactions", response_model=CashFlowTransactionResponse, status_code=status.HTTP_201_CREATED)
def create_transaction(
    transaction: CashFlowTransactionCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Criar movimentação financeira

    Registra uma entrada ou saída de caixa. Atualiza automaticamente o saldo da conta.
    """
    # Validar conta bancária
    if transaction.account_id:
        account = db.query(BankAccount).filter(
            and_(
                BankAccount.id == transaction.account_id,
                BankAccount.workspace_id == current_user.workspace_id
            )
        ).first()

        if not account:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail=f"Bank account with id {transaction.account_id} not found"
            )

    # Criar transação
    db_transaction = CashFlowTransaction(
        **transaction.dict(),
        workspace_id=current_user.workspace_id,
        created_by=current_user.id,
        updated_by=current_user.id
    )

    db.add(db_transaction)
    db.commit()
    db.refresh(db_transaction)

    # Atualizar saldo da conta
    if transaction.account_id:
        _update_account_balance(db, transaction.account_id, transaction.value, transaction.type.value, add=True)

    return db_transaction


@router.get("/transactions", response_model=List[CashFlowTransactionResponse])
def get_transactions(
    skip: int = Query(0, ge=0),
    limit: int = Query(100, ge=1, le=1000),
    start_date: Optional[date] = Query(None, description="Data inicial"),
    end_date: Optional[date] = Query(None, description="Data final"),
    type: Optional[TransactionTypeEnum] = Query(None, description="Tipo de transação"),
    category: Optional[str] = Query(None, description="Categoria"),
    account_id: Optional[int] = Query(None, description="ID da conta"),
    is_reconciled: Optional[bool] = Query(None, description="Filtrar por reconciliação"),
    search: Optional[str] = Query(None, description="Busca textual"),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Listar movimentações financeiras

    Retorna lista de transações com filtros opcionais.
    """
    query = db.query(CashFlowTransaction).filter(
        CashFlowTransaction.workspace_id == current_user.workspace_id
    )

    # Aplicar filtros
    if start_date:
        query = query.filter(CashFlowTransaction.transaction_date >= datetime.combine(start_date, datetime.min.time()))

    if end_date:
        query = query.filter(CashFlowTransaction.transaction_date <= datetime.combine(end_date, datetime.max.time()))

    if type:
        query = query.filter(CashFlowTransaction.type == type.value)

    if category:
        query = query.filter(CashFlowTransaction.category == category)

    if account_id:
        query = query.filter(CashFlowTransaction.account_id == account_id)

    if is_reconciled is not None:
        query = query.filter(CashFlowTransaction.is_reconciled == is_reconciled)

    if search:
        search_filter = f"%{search}%"
        query = query.filter(
            or_(
                CashFlowTransaction.description.ilike(search_filter),
                CashFlowTransaction.notes.ilike(search_filter)
            )
        )

    query = query.order_by(CashFlowTransaction.transaction_date.desc(), CashFlowTransaction.created_at.desc())
    transactions = query.offset(skip).limit(limit).all()

    return transactions


@router.get("/transactions/{transaction_id}", response_model=CashFlowTransactionResponse)
def get_transaction(
    transaction_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Buscar movimentação por ID

    Retorna detalhes de uma transação específica.
    """
    transaction = db.query(CashFlowTransaction).filter(
        and_(
            CashFlowTransaction.id == transaction_id,
            CashFlowTransaction.workspace_id == current_user.workspace_id
        )
    ).first()

    if not transaction:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Transaction with id {transaction_id} not found"
        )

    return transaction


@router.patch("/transactions/{transaction_id}", response_model=CashFlowTransactionResponse)
def update_transaction(
    transaction_id: int,
    transaction_update: CashFlowTransactionUpdate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Atualizar movimentação financeira

    Atualiza uma transação existente. Recalcula saldo se valor ou tipo mudarem.
    """
    db_transaction = db.query(CashFlowTransaction).filter(
        and_(
            CashFlowTransaction.id == transaction_id,
            CashFlowTransaction.workspace_id == current_user.workspace_id
        )
    ).first()

    if not db_transaction:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Transaction with id {transaction_id} not found"
        )

    # Guardar valores antigos para recalcular saldo
    old_value = db_transaction.value
    old_type = db_transaction.type
    old_account_id = db_transaction.account_id

    update_data = transaction_update.dict(exclude_unset=True)

    # Aplicar atualizações
    for field, value in update_data.items():
        setattr(db_transaction, field, value)

    db_transaction.updated_by = current_user.id
    db_transaction.updated_at = datetime.utcnow()

    # Recalcular saldo se necessário
    value_changed = 'value' in update_data and update_data['value'] != old_value
    type_changed = 'type' in update_data and update_data['type'] != old_type
    account_changed = 'account_id' in update_data and update_data['account_id'] != old_account_id

    if value_changed or type_changed or account_changed:
        # Reverter impacto da transação antiga
        if old_account_id:
            _update_account_balance(db, old_account_id, old_value, old_type, add=False)

        # Aplicar impacto da transação nova
        if db_transaction.account_id:
            _update_account_balance(db, db_transaction.account_id, db_transaction.value, db_transaction.type, add=True)

    db.commit()
    db.refresh(db_transaction)

    return db_transaction


@router.delete("/transactions/{transaction_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_transaction(
    transaction_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Deletar movimentação financeira

    Remove uma transação e atualiza o saldo da conta automaticamente.
    """
    db_transaction = db.query(CashFlowTransaction).filter(
        and_(
            CashFlowTransaction.id == transaction_id,
            CashFlowTransaction.workspace_id == current_user.workspace_id
        )
    ).first()

    if not db_transaction:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Transaction with id {transaction_id} not found"
        )

    # Reverter impacto no saldo
    if db_transaction.account_id:
        _update_account_balance(
            db,
            db_transaction.account_id,
            db_transaction.value,
            db_transaction.type,
            add=False
        )

    db.delete(db_transaction)
    db.commit()

    return None


# ============================================
# TRANSFER ENDPOINT
# ============================================

@router.post("/transfer", response_model=TransferResponse, status_code=status.HTTP_201_CREATED)
def create_transfer(
    transfer: TransferRequest,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Transferência entre contas

    Cria duas transações vinculadas: saída na conta de origem e entrada na conta de destino.
    """
    # Validar conta de origem
    from_account = db.query(BankAccount).filter(
        and_(
            BankAccount.id == transfer.from_account_id,
            BankAccount.workspace_id == current_user.workspace_id
        )
    ).first()

    if not from_account:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Source account with id {transfer.from_account_id} not found"
        )

    # Validar conta de destino
    to_account = db.query(BankAccount).filter(
        and_(
            BankAccount.id == transfer.to_account_id,
            BankAccount.workspace_id == current_user.workspace_id
        )
    ).first()

    if not to_account:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Destination account with id {transfer.to_account_id} not found"
        )

    # Criar transação de saída
    exit_transaction = CashFlowTransaction(
        workspace_id=current_user.workspace_id,
        transaction_date=transfer.transaction_date,
        type=TransactionType.SAIDA.value,
        category="Transferência",
        description=f"Transfer to {to_account.bank_name}: {transfer.description}",
        value=transfer.value,
        account_id=transfer.from_account_id,
        reference_type="transfer",
        reference_id=transfer.to_account_id,
        notes=transfer.notes,
        created_by=current_user.id,
        updated_by=current_user.id
    )

    db.add(exit_transaction)
    db.flush()  # Para obter o ID

    # Criar transação de entrada
    entry_transaction = CashFlowTransaction(
        workspace_id=current_user.workspace_id,
        transaction_date=transfer.transaction_date,
        type=TransactionType.ENTRADA.value,
        category="Transferência",
        description=f"Transfer from {from_account.bank_name}: {transfer.description}",
        value=transfer.value,
        account_id=transfer.to_account_id,
        reference_type="transfer",
        reference_id=transfer.from_account_id,
        parent_transaction_id=exit_transaction.id,
        notes=transfer.notes,
        created_by=current_user.id,
        updated_by=current_user.id
    )

    db.add(entry_transaction)
    db.commit()

    # Atualizar saldos
    _update_account_balance(db, transfer.from_account_id, transfer.value, TransactionType.SAIDA.value, add=True)
    _update_account_balance(db, transfer.to_account_id, transfer.value, TransactionType.ENTRADA.value, add=True)

    db.refresh(exit_transaction)
    db.refresh(entry_transaction)

    return TransferResponse(
        from_transaction=exit_transaction,
        to_transaction=entry_transaction
    )


# ============================================
# HELPER FUNCTIONS
# ============================================

def _update_account_balance(db: Session, account_id: int, value: float, transaction_type: str, add: bool = True):
    """Atualiza o saldo de uma conta bancária"""
    account = db.query(BankAccount).filter(BankAccount.id == account_id).first()

    if not account:
        return

    # Calcular impacto no saldo
    impact = value if transaction_type == TransactionType.ENTRADA.value else -value

    if not add:
        impact = -impact

    account.current_balance += impact
    account.updated_at = datetime.utcnow()

    db.commit()

File: admin/src/services/cash-flow.ts
import { api } from '@/lib/api';
import {
  CashFlowTransaction,
  CashFlowTransactionCreate,
  CashFlowTransactionUpdate,
  BankAccountData,
  BankAccountCreate,
  BankAccountUpdate,
  TransferRequest,
  TransferResponse,
  CashFlowSummary,
  CategorySummary,
  BalanceHistory,
  CashFlowProjectionData,
  CashFlowAnalytics,
  AccountBalanceSummary,
  CashFlowFilters
} from '@/types/financeiro';

export const cashFlowService = {
  // ==================== TRANSACTIONS ====================

  /**
   * Listar transações de fluxo de caixa com filtros
   */
  async getTransactions(params?: CashFlowFilters): Promise<CashFlowTransaction[]> {
    const queryParams = new URLSearchParams();

    if (params?.skip !== undefined) queryParams.append('skip', params.skip.toString());
    if (params?.limit !== undefined) queryParams.append('limit', params.limit.toString());
    if (params?.start_date) queryParams.append('start_date', params.start_date);
    if (params?.end_date) queryParams.append('end_date', params.end_date);
    if (params?.type) queryParams.append('type', params.type);
    if (params?.category) queryParams.append('category', params.category);
    if (params?.account_id !== undefined) queryParams.append('account_id', params.account_id.toString());
    if (params?.is_reconciled !== undefined) queryParams.append('is_reconciled', params.is_reconciled.toString());
    if (params?.min_value !== undefined) queryParams.append('min_value', params.min_value.toString());
    if (params?.max_value !== undefined) queryParams.append('max_value', params.max_value.toString());

    const query = queryParams.toString();
    const response = await api.get<CashFlowTransaction[]>(
      `/cash-flow/transactions/${query ? `?${query}` : ''}`
    );
    return response;
  },

  /**
   * Obter transação por ID
   */
  async getTransactionById(id: number): Promise<CashFlowTransaction> {
    const response = await api.get<CashFlowTransaction>(`/cash-flow/transactions/${id}`);
    return response;
  },

  /**
   * Criar nova transação
   */
  async createTransaction(data: CashFlowTransactionCreate): Promise<CashFlowTransaction> {
    const response = await api.post<CashFlowTransaction>('/cash-flow/transactions/', data);
    return response;
  },

  /**
   * Atualizar transação
   */
  async updateTransaction(
    id: number,
    data: CashFlowTransactionUpdate
  ): Promise<CashFlowTransaction> {
    const response = await api.patch<CashFlowTransaction>(
      `/cash-flow/transactions/${id}`,
      data
    );
    return response;
  },

  /**
   * Deletar transação
   */
  async deleteTransaction(id: number): Promise<void> {
    await api.delete(`/cash-flow/transactions/${id}`);
  },

  /**
   * Marcar transação como reconciliada
   */
  async reconcileTransaction(id: number): Promise<CashFlowTransaction> {
    const response = await api.post<CashFlowTransaction>(
      `/cash-flow/transactions/${id}/reconcile`,
      {}
    );
    return response;
  },

  // ==================== BANK ACCOUNTS ====================

  /**
   * Listar contas bancárias
   */
  async getBankAccounts(activeOnly: boolean = true): Promise<BankAccountData[]> {
    const queryParams = new URLSearchParams();
    if (activeOnly !== undefined) queryParams.append('active_only', activeOnly.toString());

    const query = queryParams.toString();
    const response = await api.get<BankAccountData[]>(
      `/cash-flow/bank-accounts/${query ? `?${query}` : ''}`
    );
    return response;
  },

  /**
   * Obter conta bancária por ID
   */
  async getBankAccountById(id: number): Promise<BankAccountData> {
    const response = await api.get<BankAccountData>(`/cash-flow/bank-accounts/${id}`);
    return response;
  },

  /**
   * Criar nova conta bancária
   */
  async createBankAccount(data: BankAccountCreate): Promise<BankAccountData> {
    const response = await api.post<BankAccountData>('/cash-flow/bank-accounts/', data);
    return response;
  },

  /**
   * Atualizar conta bancária
   */
  async updateBankAccount(id: number, data: BankAccountUpdate): Promise<BankAccountData> {
    const response = await api.patch<BankAccountData>(`/cash-flow/bank-accounts/${id}`, data);
    return response;
  },

  /**
   * Deletar conta bancária
   */
  async deleteBankAccount(id: number): Promise<void> {
    await api.delete(`/cash-flow/bank-accounts/${id}`);
  },

  // ==================== TRANSFERS ====================

  /**
   * Criar transferência entre contas
   */
  async createTransfer(data: TransferRequest): Promise<TransferResponse> {
    const response = await api.post<TransferResponse>('/cash-flow/transfer', data);
    return response;
  },

  // ==================== ANALYTICS ====================

  /**
   * Obter resumo de fluxo de caixa do período
   */
  async getSummary(
    startDate: string,
    endDate: string,
    accountId?: number
  ): Promise<CashFlowSummary> {
    const queryParams = new URLSearchParams();
    queryParams.append('start_date', startDate);
    queryParams.append('end_date', endDate);
    if (accountId !== undefined) queryParams.append('account_id', accountId.toString());

    const response = await api.get<CashFlowSummary>(
      `/cash-flow/analytics/summary?${queryParams.toString()}`
    );
    return response;
  },

  /**
   * Obter análise por categoria
   */
  async getByCategory(
    startDate: string,
    endDate: string,
    accountId?: number
  ): Promise<CategorySummary[]> {
    const queryParams = new URLSearchParams();
    queryParams.append('start_date', startDate);
    queryParams.append('end_date', endDate);
    if (accountId !== undefined) queryParams.append('account_id', accountId.toString());

    const response = await api.get<CategorySummary[]>(
      `/cash-flow/analytics/by-category?${queryParams.toString()}`
    );
    return response;
  },

  /**
   * Obter resumo por conta bancária
   */
  async getByAccount(): Promise<AccountBalanceSummary[]> {
    const response = await api.get<AccountBalanceSummary[]>(
      '/cash-flow/analytics/by-account'
    );
    return response;
  },

  /**
   * Obter histórico de saldo diário
   */
  async getBalanceHistory(
    startDate: string,
    endDate: string,
    accountId?: number
  ): Promise<BalanceHistory[]> {
    const queryParams = new URLSearchParams();
    queryParams.append('start_date', startDate);
    queryParams.append('end_date', endDate);
    if (accountId !== undefined) queryParams.append('account_id', accountId.toString());

    const response = await api.get<BalanceHistory[]>(
      `/cash-flow/analytics/balance-history?${queryParams.toString()}`
    );
    return response;
  },

  /**
   * Obter projeção de fluxo de caixa
   */
  async getProjection(
    daysAhead: number = 30,
    accountId?: number
  ): Promise<CashFlowProjectionData[]> {
    const queryParams = new URLSearchParams();
    queryParams.append('days_ahead', daysAhead.toString());
    if (accountId !== undefined) queryParams.append('account_id', accountId.toString());

    const response = await api.get<CashFlowProjectionData[]>(
      `/cash-flow/analytics/projection?${queryParams.toString()}`
    );
    return response;
  },

  /**
   * Obter analytics completo
   */
  async getCompleteAnalytics(
    startDate: string,
    endDate: string,
    accountId?: number
  ): Promise<CashFlowAnalytics> {
    const queryParams = new URLSearchParams();
    queryParams.append('start_date', startDate);
    queryParams.append('end_date', endDate);
    if (accountId !== undefined) queryParams.append('account_id', accountId.toString());

    const response = await api.get<CashFlowAnalytics>(
      `/cash-flow/analytics/complete?${queryParams.toString()}`
    );
    return response;
  },

  // ==================== BATCH OPERATIONS ====================

  /**
   * Marcar múltiplas transações como reconciliadas
   */
  async batchReconcile(
    ids: number[]
  ): Promise<{ success: number; failed: number; results: CashFlowTransaction[] }> {
    const response = await api.post<{
      success: number;
      failed: number;
      results: CashFlowTransaction[];
    }>('/cash-flow/transactions/batch/reconcile', { ids });
    return response;
  },

  /**
   * Deletar múltiplas transações
   */
  async batchDelete(ids: number[]): Promise<{ success: number; failed: number }> {
    const response = await api.post<{ success: number; failed: number }>(
      '/cash-flow/transactions/batch/delete',
      { ids }
    );
    return response;
  },

  /**
   * Exportar transações
   */
  async export(
    format: 'csv' | 'xlsx' | 'pdf',
    filters?: CashFlowFilters
  ): Promise<Blob> {
    const queryParams = new URLSearchParams();
    queryParams.append('format', format);

    if (filters?.start_date) queryParams.append('start_date', filters.start_date);
    if (filters?.end_date) queryParams.append('end_date', filters.end_date);
    if (filters?.type) queryParams.append('type', filters.type);
    if (filters?.category) queryParams.append('category', filters.category);
    if (filters?.account_id !== undefined) queryParams.append('account_id', filters.account_id.toString());

    const query = queryParams.toString();
    const response = await api.get<Blob>(
      `/cash-flow/export${query ? `?${query}` : ''}`,
      { responseType: 'blob' }
    );
    return response;
  }
};
